[ONNX] Update ONNX to rel-1.9 (#55889)

neginraoof opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57080 [ONNX] Update ONNX to rel-1.9 (#55889)**
* #57079 [ONNX] Handle PackedParams inputs for _propagate_and_assign_input_shapes (#56449)

ONNX optimizer is deprecated in ONNX 1.9 and  being removed from caffe2 import module by this PR.
Co-authored-by: neginraoof <neginmr@utexas.edu>

Differential Revision: [D28467330](https://our.internmc.facebook.com/intern/diff/D28467330)

neginraoof committed:
[ONNX] Update ONNX to rel-1.9 (#55889)

ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

malfet commented on caffe2/CMakeLists.txt:
```diff
@@ -141,6 +141,17 @@ if(BUILD_CAFFE2 AND (NOT INTERN_BUILD_MOBILE OR BUILD_CAFFE2_MOBILE))
   add_subdirectory(observers)
   add_subdirectory(onnx)
+
+  include_directories(../third_party/optimizer)
+
+  set(ONNX_SRCS
+    "../third_party/optimizer/onnxoptimizer/optimize.cc"
+    "../third_party/optimizer/onnxoptimizer/pass.cc"
+    "../third_party/optimizer/onnxoptimizer/pass_registry.cc"
+    "../third_party/optimizer/onnxoptimizer/pass_manager.cc")
+
+  list(APPEND Caffe2_CPU_SRCS ${ONNX_SRCS})
```
Please move this code to `cmake/Dependencies.cmake` next to the other ONNX dependencies logic.
And is there a reason why we need to explicitly specify source files rather than just make caffe2 depend on  `onnx_optimizer` library?

neginraoof replied:
@malfet 
Thanks, I'm trying to include the Optimizer library in the Dependencies.cmake. But I'm not sure how to make this work where USE_SYSTEM_ONNX is set. How can we make sure Optimizer is set up in such environments?

neginraoof replied:
@malfet 
The onnx optimizer library also depends on ONNX and this is causing a dependency cycle. What do you think?

malfet replied:
Let's not worry about `USE_SYSTEM_ONNX` for the time being. But I'm not sure what do you mean about cyclic dependency? This looks like a DAG to me, isn't i, i.e. libtorch_cpu depends on onnx and onnxoptimizer, while onnxoptimizer depends on onnx,  i.e.:
```
 libtorchcpu <--- ONNX
     ^             ^
     |             |
     |---- ONNXoptimizer
```

neginraoof replied:
So ONNX is a submodule of the optimizer. In this case, wouldn't it look like:
```
 libtorchcpu +--- ONNX
     +           |
     |           +
     |---- ONNXoptimizer

malfet reviewed (changes requested):
Please move dependencies logic to https://github.com/pytorch/pytorch/blob/2a178d34cd2acd04c05050689a25c79e777c28c5/cmake/Dependencies.cmake#L1463

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

malfet commented:
Can you please rebase your change against `viable/strict`, otherwise it's very hard to distinguish between actual failures and false positives.

malfet commented on cmake/Dependencies.cmake:
```diff
@@ -1482,7 +1480,17 @@ if(CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO AND NOT INTERN_DISABLE_ONNX)
+
+  add_definitions(-DONNX_NAMESPACE=${ONNX_NAMESPACE})
+  include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/third_party/optimizer)
+  add_library(onnx_optimizer STATIC
```
Why re-defining `onnx_optimizer` library here instead of using one defined in (which can be included using `add_subdirectory("${CAFFE2_THIRD_PARTY_ROOT}/onnx_optimizer" "${CONFU_DEPENDENCIES_BINARY_DIR}/onnx_optimizer")`  https://github.com/onnx/optimizer/blob/a37748b2c3a80dad4274401c45c5026c7a506730/CMakeLists.txt#L23

neginraoof replied:
Updating this, now I see the error:
 add_subdirectory given source
  "/var/lib/jenkins/workspace/third_party/onnx_optimizer" which is not an
  existing directory.

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

neginraoof committed:
Update on "[ONNX] Update ONNX to rel-1.9 (#55889)"


ONNX optimizer is deprecated in ONNX 1.9 and needs to be included in Caffe2 as a submodule.

Co-authored-by: neginraoof <neginmr@utexas.edu>

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28467330).

malfet commented:
This PR caused some regressions on internal testsuite, investigating...

neginraoof commented:
Caffe2 failures could be expected due to deprecation of the optimizer. Please let us know about the failing tests.

neginraoof commented:
@malfet 
Do you have any updates regarding failing tests?

neginraoof commented:
@malfet
Do you have any updates on disable the failing tests?

malfet commented:
@neginraoof sorry for the late replies. Modified caffe2/python/onnx/backend.py to keep optimizer as fallback dependency to workaround test failures. Seems to be passing all the tests now.
